### ISSUE TITLE
Recieved video message from ios device is showing corrupted

### ISSUE THREAD

**Issue by rekhaagarwal** [layerhq/Atlas-Android#160]
Hello
I am using atlas version: 0.4.5. As per my knowledge, atlas android sdk is not supporting video message for now but ios does. Hence while chatting with ios user, when we get video message then on latest version its currently being displayed as below image.
in earlier atlas version, say 0.3.0, it was being handled very gracefully . we were getting a image part instead earlier. so is there any changes in latest release and how to deal with it.
We are using AtlasCellFactory for showing image/video cell. In current version, the video message on android is assigned GenericCellfactory.

![screenshot_20170316-054344](https://cloud.githubusercontent.com/assets/21167009/24031212/5bec43d8-0b07-11e7-9969-a01430001011.png)

**Comment by andela-wakinsanmi** [comment]
Hello @rekhaagarwal 

To handle videos, you will need to implement your own CellFactory to support MimeType video/mp4.

Can you please provide the CellFactory in 0.3.0 that handles video gracefully. The CellFactories that are available in 0.3.0 will check if the first message type has MimeType starting with image, using the following condition `getMimeType().startsWith("image/")`.

The message type above will be handled by the Generic cell factory.


Thanks

**Comment by sriamar** [comment]
Hello,

To be specific, we have made the matching strict in our message part handlers. We used to be very lenient on what we match to `SinglePartImageCellFactory` as shown here.
https://github.com/layerhq/Atlas-Android/pull/124/files#diff-44cb3632df57f849bce9e97e8e97e520L121

With the change here, we have made it strict
https://github.com/layerhq/Atlas-Android/pull/124/files#diff-44cb3632df57f849bce9e97e8e97e520R114

If you are still blocked by this, let us know and we can suggest you alternatives.

cc : @thecombatwombat
